metrics/report/report_dockerfile/collectd_scaling.R

```diff
@@ -95,6 +99,41 @@ for (currentdir in resultdirs) {
 			local_bootdata$epoch = local_bootdata$ns/1000000000
 			local_bootdata$s_offset = local_bootdata$epoch - local_bootdata[1,]$epoch
 
+			# Grab the stepping used during testing, so we can calculate bucket sizes
+			stepping=fdata$Config$STEP
+
+			# Here we try to calculate and get a feel for 'changes over time' for
+			# launch times. That is, we are looking to see if the 'launch overheads
+			# per pod' change over time (or is it a linear cost?).
+			# We do this by taking the launch time diff between each consequtive launch,
+			# and then looking at the mean of those diffs across a 'buckets' width. We do
+			# this to de-noise the data (as actual individual diffs bounce up and down around
+			# the zero line a lot... so we are looking for the 'trend')
+
+			# Set up a diff of the consequtive boot times. Prepend with '0' so the result
+			# is the same length as the original data...
+			local_bootdata$launch_diff=c(0, diff(local_bootdata$launch_time))
+
+			# Annotate entries into 'buckets', so we can get some 'rolling' means
+			# Calculate the bucket mid point values of the range they cover, so they
+			# plot sanely along with n_pods based data.
+			bucket_size=ceiling(nrow(local_bootdata)/bootdiffbuckets)
+
+			# Take into account any STEP used in the testing.
+			pods_per_bucket=bucket_size*stepping
+			local_bootdata$bucket_min=(rep(0:(bootdiffbuckets-1), each=bucket_size)*bucket_size)[1:nrow(local_bootdata)] * stepping
+			local_bootdata$bucket_max=local_bootdata$bucket_min+(pods_per_bucket-1)
+			local_bootdata$bucket_mid=local_bootdata$bucket_min + (pods_per_bucket/2)
```

Comment:
nitpicky: user (or not) space between main addition operation.

Reply:
ack. good spot. went with spaces for this patch.